gnomad_tgps.py

```diff
@@ -0,0 +1,86 @@
+import hail as hl
+from gnomad_qc.v3.resources import basics
+import hailtop.fs as hfs
+
+# initialize Hail
+hl.init(log='test.log', default_reference='GRCh38', backend='spark')
+
+# get the site table to extract AC and AF frequencies
+gnomad_freq_path = "gs://gcp-public-data--gnomad/release/3.1.2/ht/genomes/gnomad.genomes.v3.1.2.sites.ht"
+gnomad_freq_ht = hl.read_table(gnomad_freq_path)
+
+# get gnomad V3 data and split multi-allelic variants
+vds = basics.get_gnomad_v3_vds()
+vds_mt = vds.variant_data
+vds_split = hl.experimental.sparse_split_multi(vds_mt)
+
+# get the metadata of gnomad v3
+meta_path = "gs://gnomad/metadata/genomes_v3.1/gnomad_v3.1_sample_qc_metadata.ht"
+meta_dat = hl.read_table(meta_path)
+
+# filter out unreleased samples
+release_ht = meta_dat.filter(meta_dat.release == True)
+vds_split_filterd = vds_split.filter_cols(hl.is_defined(release_ht[vds_split.col_key]))
+
+# load samples in the pangenome
+mt1kg = vds_split_filterd
+table_1kg = meta_dat
+
+p1_ids = []
+id_path = "gs://jialan-tmp-7day/sample_ids_phase2_n55_all.txt"  # read the sample IDs for the reference set
+with hfs.open(id_path, 'r') as p1_file:
+    for line in p1_file:
+        sid = line.strip()
+        p1_ids.append(f'{sid}_{sid}')
+p1_ids = hl.set(p1_ids)
+
+# Divide the table with genetic data into sample (reference) and out-of-sample populations
+mt1kg_oos = mt1kg.filter_cols(p1_ids.contains(mt1kg.s), keep=False)
+mt1kg_s = mt1kg.filter_cols(p1_ids.contains(mt1kg.s), keep=True)
```

Comment:
it might be slightly cleaner to do this:
```
mt = mt.annotate(s_in_pangenome=p1_ids.contains(mt.s))
pg_mt = mt.filter_cols(mt.s_in_pangenome)
mt = mt.filter_cols(~mt.s_in_pangenome)
```
since the set will only get compared to the samples in your input MT once